src/libgap-api.c

```diff
@@ -313,3 +314,11 @@ Obj GAP_CharWithValue(UChar obj)
 {
     return ObjsChar[obj];
 }
+
+double GAP_ValueMacFloat(Obj obj)
+{
+    if (!IS_MACFLOAT(obj)) {
```

Comment:
On the assumption that interface users might not like ErrorQuits, how about an interface:

`Int GAP_ValueMacFloat(Obj obj, double* d);`

Write value of obj into d, return 0 on failure, 1 on success, inspired by scanf and friends.

Reply:
I am not so fond of the idea. Now we've replaced one source of mistakes with another: what if you pass in an invalid pointer? OK, we can at least add a check `x != 0`, but that won't catch references to stack values that already went out of range and silliness like that.

All code I know which would call something like (GAPJulia, SingularInterface, NormalizInterface, SageMath interface for GAP) actually looks roughly like the following anyway, rendering the extra check unnecessary (and in fact, inconvenient):
```C
  if (GAP_IsInt(obj)) {
    return convert_gap_int_to_our_format(obj);
  }
  else if (GAP_IsMacFloat(obj)) {
    return convert_double_to_our_format( GAP_ValueMacFloat(obj) );
  }
  ...
```

But if we really think we should use this kind of indirection here, then I think we should be consistent and do it for all the other getters we already defined resp. will define in the future.